src/test/java/org/opensearch/security/auditlog/compliance/ComplianceAuditlogTest.java

```diff
@@ -183,13 +183,12 @@ public void testSourceFilterMsearch() throws Exception {
                 "   }" +
                 "}"+System.lineSeparator();
 
-        TestAuditlogImpl.clear();
-        HttpResponse response = rh.executePostRequest("_msearch?pretty", search, encodeBasicHeader("admin", "admin"));
-        assertNotContains(response, "*exception*");
-        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
-        Thread.sleep(1500);
+        TestAuditlogImpl.doThenWaitForMessages(() -> {
```

Comment:
👏

Reply:
😄

Reply:
However, we still have many places we could apply these changes.  Maybe another time we will do a pass to ensure reliability of tests